deployments/ecmps/release-v2.0-fix/Sprint8/6080/camdecmps/get-qatest-feedback.sql

```diff
@@ -0,0 +1,36 @@
+DROP FUNCTION IF EXISTS camdecmps.get_qatest_feedback(text) CASCADE;
+
+CREATE OR REPLACE FUNCTION camdecmps.get_qatest_feedback(testId text)
+RETURNS TABLE(
+    unit_stack_pipe character varying,
+    test_type_cd character varying,
+    test_num character varying,
+    sys_comp text,
+    end_test character varying)
+LANGUAGE plpgsql
+AS $$
+BEGIN
+RETURN QUERY
+SELECT
+    COALESCE(ml.stack_name, ml.unitid) AS unit_stack_pipe,
+    ts.test_type_cd,
+    ts.test_num,
+    COALESCE(c.component_identifier, ms.system_identifier, '') || '/' ||
+    COALESCE(c.component_type_cd || ' ' || COALESCE(ts.span_scale_cd, ''), ms.sys_type_cd, '') AS sys_comp,
+    CASE
+        WHEN rp.calendar_year IS NOT NULL THEN rp.period_abbreviation
+        ELSE camdecmps.format_date_hour(ts.end_date, ts.end_hour, ts.end_min)
+        END AS end_test
+FROM camdecmps.test_summary ts
+         INNER JOIN (
```

Comment:
The cl_filtered sub-query is not actually used in the query.  Is the purpose to make sure that a CHECK_LOG row exists for the associated Check Session?  If so, this method could use to duplicate rows for when more than one CHECK_LOG row exists.

Reply:
So, basically I replicated the existing query from SQL Server you provided to me. 

The relevant section of the original SQL Server  query is as follows (redacted)

FROM VW_RPT_FEEDBACK_CHECK_LOG cl
			INNER JOIN TEST_SUMMARY ts ON cl.CHK_SESSION_ID = ts.CHK_SESSION_ID 
			INNER JOIN QA_SUPP_DATA qa ON ts.TEST_SUM_ID = qa.TEST_SUM_ID 
			INNER JOIN VW_MONITOR_LOCATION ml ON ts.MON_LOC_ID = ml.MON_LOC_ID 
			LEFT OUTER JOIN COUNTY_CODE cc ON ml.COUNTY_CD = cc.COUNTY_CD 
			LEFT OUTER JOIN REPORTING_PERIOD rp ON ts.RPT_PERIOD_ID = rp.RPT_PERIOD_ID 
			LEFT OUTER JOIN COMPONENT c ON ts.COMPONENT_ID = c.COMPONENT_ID 
			LEFT OUTER JOIN MONITOR_SYSTEM ms ON ts.MON_SYS_ID = ms.MON_SYS_ID 
		WHERE (cl.SEVERITY_CD <> 'NONE' OR cl.SEVERITY_CD IS NULL)
		  AND ts.TEST_SUM_ID IN

I basically moved the VW_RPT_FEEDBACK_CHECK_LOG  FROM clause to an inner join. Note also that  this WHERE check that is in the inner join from the VW_RPT_FEEDBACK_CHECK_LOG  
   WHERE cl.severity_cd <> 'NONE' OR cl.severity_cd IS NULL
   
You also stated that... "If so, this method could use to duplicate rows for when more than one CHECK_LOG row exists."

I am not sure the implication of this, but my goal was to replicate the SQL Server logic as accurately as possible.

Reply:
Thanks for the clarification and I agree with your goal.  I am going to resolve this conversation.